gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/EntityExecutionContext.java

```diff
@@ -0,0 +1,228 @@
+      EntityIdColumnsConfigs entityIdColumnsConfigs,
+      EntitiesRequestContext entitiesRequestContext,
+      EntitiesRequest entitiesRequest) {
+    this.attributeMetadataProvider = attributeMetadataProvider;
+    this.entityIdColumnsConfigs = entityIdColumnsConfigs;
+    this.entitiesRequest = entitiesRequest;
+    this.entitiesRequestContext = entitiesRequestContext;
+
+    Map<String, AttributeMetadata> attributeMetadataMap =
+        attributeMetadataProvider.getAttributesMetadata(
+            this.entitiesRequestContext, entitiesRequestContext.getEntityType());
+    this.expressionContext =
+        new ExpressionContext(
+            attributeMetadataMap,
+            entitiesRequest.getFilter(),
+            entitiesRequest.getSelectionList(),
+            entitiesRequest.getTimeAggregationList(),
+            entitiesRequest.getOrderByList(),
+            // entities request does not have group by
+            Collections.emptyList());
+
+    buildSelectionPendingSources();
+    buildOrderByPendingSources();
+  }
+
+  public String getTenantId() {
+    return entitiesRequestContext.getTenantId();
+  }
+
+  public String getTimestampAttributeId() {
+    return entitiesRequestContext.getTimestampAttributeId();
+  }
+
+  public AttributeMetadataProvider getAttributeMetadataProvider() {
+    return attributeMetadataProvider;
+  }
+
+  public Map<String, String> getRequestHeaders() {
+    return entitiesRequestContext.getHeaders();
+  }
+
+  public EntitiesRequest getEntitiesRequest() {
+    return entitiesRequest;
+  }
+
+  public EntitiesRequestContext getEntitiesRequestContext() {
+    return this.entitiesRequestContext;
+  }
+
+  public ExpressionContext getExpressionContext() {
+    return this.expressionContext;
+  }
+
+  public Set<String> getPendingSelectionSources() {
+    return pendingSelectionSources;
+  }
+
+  public Set<String> getPendingMetricAggregationSources() {
+    return pendingMetricAggregationSources;
+  }
+
+  public Set<String> getPendingTimeAggregationSources() {
+    return pendingTimeAggregationSources;
+  }
+
+  public Set<String> getPendingSelectionSourcesForOrderBy() {
+    return pendingSelectionSourcesForOrderBy;
+  }
+
+  public Set<String> getPendingMetricAggregationSourcesForOrderBy() {
+    return pendingMetricAggregationSourcesForOrderBy;
+  }
+
+  public boolean isSortAndPaginationNodeAdded() {
+    return sortAndPaginationNodeAdded;
+  }
+
+  public void setSortAndPaginationNodeAdded(boolean sortAndPaginationNodeAdded) {
+    this.sortAndPaginationNodeAdded = sortAndPaginationNodeAdded;
+  }
+
+  public List<Expression> getEntityIdExpressions() {
+    List<String> entityIdAttributeNames =
+        AttributeMetadataUtil.getIdAttributeIds(
+            attributeMetadataProvider,
+            entityIdColumnsConfigs,
+            this.entitiesRequestContext,
+            entitiesRequestContext.getEntityType());
+    return IntStream.range(0, entityIdAttributeNames.size())
```

Comment:
nit: why not just `entityIdAttributeNames.stream().map(...).collect(Collectors.toList())`